pkg/flagutil/flagutil_test.go

```diff
@@ -73,3 +97,223 @@ func TestReplaceOrAppendEnvValues(t *testing.T) {
 		assert.Assert(t, comparator(actual, tt.expected), fmt.Sprintf("expected: %s, actual: %s", tt.expected, actual))
 	}
 }
+
+// Test TestParseEnvFileGoodFile for a env file with a few well formatted lines.
+func TestParseEnvFileGoodFile(t *testing.T) {
+	content := `foo=bar
+    baz=quux
+# comment
+
+_foobar=foobaz
+with.dots=working
+and_underscore=working too`
+	// Adding a newline + a line with pure whitespace.
+	// This is being done like this instead of the block above
+	// because it's common for editors to trim trailing whitespace
+	// from lines, which becomes annoying since that's the
+	// exact thing we need to test.
+	content += "\n    \t  "
+	tmpFile := tmpFileWithContent(t, content)
+
+	lines, err := parseEnvVars([]string{tmpFile})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expectedLines := []string{
+		"foo=bar",
+		"baz=quux",
+		"_foobar=foobaz",
+		"with.dots=working",
+		"and_underscore=working too",
+	}
+
+	if !reflect.DeepEqual(lines, expectedLines) {
+		t.Fatal("lines not equal to expectedLines")
+	}
+}
+
+// Test TestParseEnvFileEmptyFile for an empty file.
+func TestParseEnvFileEmptyFile(t *testing.T) {
+	tmpFile := tmpFileWithContent(t, "")
+
+	paths := []string{tmpFile}
+	lines, err := parseEnvVars(paths)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(lines) != 0 {
+		t.Fatal("lines not empty; expected empty")
+	}
+}
+
+// Test TestParseEnvFileNonExistentFile for a non existent file.
+func TestParseEnvFileNonExistentFile(t *testing.T) {
+	_, err := parseEnvVars([]string{"foo_bar_baz"})
+	if err == nil {
+		t.Fatal("ParseEnvFile succeeded; expected failure")
+	}
+	if _, ok := errors.Unwrap(err).(*os.PathError); !ok {
+		t.Fatalf("Expected a PathError, got [%v]", err)
+	}
+}
+
```

Comment:
is it possible to merge  `TestParseEnvFileGoodFile` , `TestParseEnvFileEmptyFile` and `TestParseEnvFileNonExistentFile` and use cases instead

Reply:
I think the current way of doing tests is more readable.